op-chain-ops/deployer/integration_test/apply_test.go

```diff
@@ -261,10 +267,31 @@ func validateOPChainDeployment(t *testing.T, ctx context.Context, l1Client *ethc
 
 		t.Run("l2 genesis", func(t *testing.T) {
 			require.Greater(t, len(chainState.Allocs), 0)
+			l2Allocs, _ := chainState.UnmarshalAllocs()
+			alloc := l2Allocs.Copy().Accounts
+
+			checkImmutable(t, alloc, "BaseFeeVault", "0xC0d3c0D3c0d3C0D3C0D3C0d3c0D3C0D3c0d30019", st.AppliedIntent.BaseFeeVaultRecipient.Hex())
+			checkImmutable(t, alloc, "L1FeeVault", "0xc0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d3001a", st.AppliedIntent.L1FeeVaultRecipient.Hex())
+			checkImmutable(t, alloc, "SequencerFeeVault", "0xC0D3C0d3c0d3c0d3C0D3c0d3C0D3c0d3c0D30011", st.AppliedIntent.SequencerFeeVaultRecipient.Hex())
+
+			require.Equal(t, int(st.AppliedIntent.Eip1559Denominator), 50, "EIP1559Denominator should be set")
+			require.Equal(t, int(st.AppliedIntent.Eip1559Elasticity), 6, "EIP1559Elasticity should be set")
 		})
 	}
 }
 
+func checkImmutable(t *testing.T, accounts types.GenesisAlloc, vaultName, vaultAddress, recipientHex string) {
+	account, ok := accounts[common.HexToAddress(vaultAddress)]
+	require.True(t, ok, "%s at %s not found in allocations", vaultName, vaultAddress)
+	require.NotEmpty(t, account.Code, "%s Impl should have code", vaultName)
+	require.Contains(
```

Comment:
No doubt this is super messy but I'm not sure of a better way to check that the immutable was set when only given the genesis file. 

I also don't like having to explicitly use the `0xc0d3...` addresses.

Reply:
Agreed that would be a stronger test, but I'm unsure how to do that, and this is probably sufficient because the odds the bytecode contains the same 20 byte string as the specified address is very small